### ISSUE TITLE
Wrapping environment variables in GNOME 3 applications

### ISSUE THREAD

**Issue by romildo** [NixOS/nixpkgs#14120]
It seems that all nix expressions for gnome3 applications need revision regarding environment variables wrapping.

For instance  on my unstable NixOS system I have

```
    gnome3.dconf-editor
    gnome3.eog
    gnome3.epiphany
    gnome3.evince
    gnome3.file-roller
    gnome3.gedit
    gnome3.gnome-calculator
    gnome3.gnome_terminal
    gnome3.nautilus
```

on `environment.systemPackages` on my configuration. All of them (except `gnome-terminal`, which I have already fixed), when run outside of gnome3, are giving error messages like

```
GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
```

I think they may be fixed by replacing uses of `wrapProgram` to explicitly fix binaries, by `wrapGAppsHook`, which automatically fixes them wrapping environment variables.

EDIT: In some cases it also needs to add a dependency on `gnome3.dconf`.

**Comment by jgillich** [comment]
Related: #4415

**Comment by abbradar** [comment]
cc @lethalman

**Comment by jtojnar** [comment]
I think  the last instances of this were fixed by #26113

**Comment by jtojnar** [comment]
In the core GNOME, I only came across two not wrapped executables. `gsettings` from `glib` package and `at-spi-bus-launcher` from `at-spi2-core`. Unfortunately the former cannot be wrapped due to `dconf` depending on `glib`.
